Entity/OverrideLeadFieldRepository.php

```diff
@@ -212,29 +221,46 @@ public function compareValue($lead, $field, $value, $operatorExpr)
     }
 
     /**
-     * Get an extended field given the field alias.
-     *
-     * @param string $alias
+     * @param $fieldAlias
      *
-     * @return null|array
+     * @return array|bool|mixed|null
      */
-    private function getExtendedField($alias)
+    private function getExtendedFieldConfigurations($fieldAlias)
     {
-        $qf = $this->_em->getConnection()->createQueryBuilder();
-        $qf->select('lf.id, lf.object, lf.type, lf.alias, lf.field_group as "group", lf.object, lf.label')
-            ->from(MAUTIC_TABLE_PREFIX.'lead_fields', 'lf')
-            ->where(
-                $qf->expr()->andX(
-                    $qf->expr()->eq('lf.alias', ':alias'),
-                    $qf->expr()->orX(
-                        $qf->expr()->eq('lf.object', $qf->expr()->literal('extendedField')),
-                        $qf->expr()->eq('lf.object', $qf->expr()->literal('extendedFieldSecure'))
-                    )
-                )
-            )
-            ->setParameter('alias', $alias);
+        if (!$this->extendedFieldConfigurations) {
+            $cacheHelper     = new CacheStorageHelper(
+                CacheStorageHelper::ADAPTOR_FILESYSTEM,
+                'extendedFieldConfiguration',
+                null,
+                $this->coreParametersHelper->getParameter('mautic.cache_path'),
+                60
+            );
```

Comment:
Ideally I'd prefer if CacheStorageHelper was passed through the constructor, so it's more configurable to the consumer of the class.

Reply:
Normally, yes. But this is an override class, and it'll never be consumed in such a way. Best to keep the constructor as stock as possible to reduce collisions when core changes.